File: nightly/tests/java/Reassignment.java
// TODO this test case needs "unexpected_lines" 7
public final class Reassignment {
    static String source() { return "tainted"; }
    static void sink(String s) { System.out.println(s); }

    public static void main(String[] args) {
        String data = source(); // Line 7
        data = "safe";          // Line 8
        sink(data);             // Line 9
    }
}

File: nightly/tests/python/reassignment.py
# Negative case: the tainted value is overwritten before it reaches the sink, so
# no flow should be reported. source() taints `data`, but `data` is reassigned
# to a constant on the next line; the sink therefore sees only clean data.
# Empty expected_lines makes this a negative test.
def main():
    data = source()
    data = "clean"
    sink(data)
